
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../20-short-read-qc/">
      
      
        <link rel="next" href="../50-assembly-qc-exercise/">
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.0, mkdocs-material-9.5.25">
    
    
      
        <title>Quality control criteria for genome assemblies - The Microbial Bioinformatics Handbook</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.6543a935.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#contiguity-genome-assembly" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="The Microbial Bioinformatics Handbook" class="md-header__button md-logo" aria-label="The Microbial Bioinformatics Handbook" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            The Microbial Bioinformatics Handbook
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Quality control criteria for genome assemblies
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var media,input,key,value,palette=__md_get("__palette");if(palette&&palette.color){"(prefers-color-scheme)"===palette.color.media&&(media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']"),palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent"));for([key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="The Microbial Bioinformatics Handbook" class="md-nav__button md-logo" aria-label="The Microbial Bioinformatics Handbook" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    The Microbial Bioinformatics Handbook
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Welcome to The Microbial Bioinformatics Handbook
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../glossary/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Glossary
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Assembly compare
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            Assembly compare
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../assembly-compare/assembly-hybrid-short-read/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Hybrid and short-read-only assembly of sequenced reads
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Concepts
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            Concepts
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../concepts/fastq-in-detail/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    FASTQ in detail
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../concepts/file-formats/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Exploring file formats
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../concepts/genome-assembly/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    What is a genome assembler doing?
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../concepts/sequence-alignment/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Local vs global alignment
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../concepts/sequence-data/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Revisting sequence file formats
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../concepts/sequencing-reading/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Background to sequencing
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Exercise answers
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Exercise answers
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../exercise-answers/check-qc-answers/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Answeres for Practical - Genome assembly QC
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../exercise-answers/fastq-in-detail-answers/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    FASTQ in detail answers
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../exercise-answers/file-compression-answers/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    File compression answers
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../exercise-answers/file-formats-answers/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Common file formats (answers)
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../exercise-answers/long-read-qc-answers/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Long read qc answers
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../exercise-answers/read-class-answers/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Answers for read classification
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../exercise-answers/sequence-data-answers/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Sequence data answers
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../exercise-answers/short-read-qc-answers/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Answers for read QC - FASTQC
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../exercise-answers/short-read-qc-answers2/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Answers for read QC - trimming
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../exercise-answers/using-gzip-answers/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Answers for Using tar and gzip
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Genepi biotrain
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            Genepi biotrain
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../genepi-biotrain/00-welcome/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Welcome!
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../genepi-biotrain/90-further-reading/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Further reading and additional resources
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7" >
        
          
          <label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Genome annotation
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7">
            <span class="md-nav__icon md-icon"></span>
            Genome annotation
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../genome-annotation/2000-01-07-annotation/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Genome browsers and annotation
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../genome-annotation/2000-01-08-auto-annotation/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Automated annotation
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_8" checked>
        
          
          <label class="md-nav__link" for="__nav_8" id="__nav_8_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Quality control
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_8_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_8">
            <span class="md-nav__icon md-icon"></span>
            Quality control
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../00-poor-qc-issues/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Why is quality control important?
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../01-qc-framework/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    A framework for quality control
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../05-read-qc/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Quality control criteria for sequenced reads
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../10-read-classification/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Practical - Read classification of our sequenced data
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../11-read-classification-cmd/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Practical - Read classification (command-line)
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../20-short-read-qc/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Practical - Quality control for short reads
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Quality control criteria for genome assemblies
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Quality control criteria for genome assemblies
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#contiguity-genome-assembly" class="md-nav__link">
    <span class="md-ellipsis">
      Contiguity (Genome assembly)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#completeness-genome-assembly" class="md-nav__link">
    <span class="md-ellipsis">
      Completeness (Genome assembly)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#contamination-genome-assembly" class="md-nav__link">
    <span class="md-ellipsis">
      Contamination (Genome assembly)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#correctness-genome-assembly" class="md-nav__link">
    <span class="md-ellipsis">
      Correctness (Genome assembly)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#bonus-circumstantial-genome-assembly" class="md-nav__link">
    <span class="md-ellipsis">
      BONUS: Circumstantial (Genome assembly)
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../50-assembly-qc-exercise/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Practical - Genome assembly QC
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_9" >
        
          
          <label class="md-nav__link" for="__nav_9" id="__nav_9_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Unix
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_9_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_9">
            <span class="md-nav__icon md-icon"></span>
            Unix
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../unix/file-compression/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Understanding file compression
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../unix/using-gzip/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Using tar and gzip for fun and profit
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#contiguity-genome-assembly" class="md-nav__link">
    <span class="md-ellipsis">
      Contiguity (Genome assembly)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#completeness-genome-assembly" class="md-nav__link">
    <span class="md-ellipsis">
      Completeness (Genome assembly)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#contamination-genome-assembly" class="md-nav__link">
    <span class="md-ellipsis">
      Contamination (Genome assembly)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#correctness-genome-assembly" class="md-nav__link">
    <span class="md-ellipsis">
      Correctness (Genome assembly)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#bonus-circumstantial-genome-assembly" class="md-nav__link">
    <span class="md-ellipsis">
      BONUS: Circumstantial (Genome assembly)
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  <h1>Quality control criteria for genome assemblies</h1>

<p>Go back to <a href="../01-qc-framework/">A framework for quality control</a></p>
<p>Quality control processes for genome assemblies aim to answer "<em>Does the genome assembly look like an intact genome from the organism I am expecting?</em>" The criteria Contiguity, Completeness, Contamination, Correctness assess this in different ways. </p>
<p>The ultimate goal of genome assembly is to reconstruct the original genome sequence, but to do this we have sheared, amplified the DNA and read off short sequences. There are errors introduced at each step. At the time of writing, there is no reliable method of automatically generating a perfect and complete genome sequence. We must make do with what we have, and be able to assess if the genome assembly is 'good enough' for our purposes.</p>
<p>There are many tools available to assess genome assembly quality. We will look at some of the most common ones. </p>
<h2 id="contiguity-genome-assembly">Contiguity (Genome assembly)</h2>
<p>As mentioned above, we are aiming, but not expecting, a complete single genome (chromosome) sequence. Contiguity measures how contiguous the assembled genome is, the less fragements the better. You can look at metrics like the <code>N50</code> and <code>L50</code>, which indicate the length of the longest contig and the number of contigs needed to cover a certain percentage of the genome. Higher <code>N50</code> and lower <code>L50</code> values are generally better. How can we assess contiguity?</p>
<ul>
<li>Less contigs, Longer contigs</li>
<li>N50, average contig length, number of contigs etc.</li>
<li>Try <a href="https://quast.sourceforge.net/quast.html">QUAST</a></li>
</ul>
<div class="admonition tip">
<p class="admonition-title">What is a contig?</p>
<p>A "contig" (short for contiguous sequence) is a set of overlapping DNA segments that together represent a consensus region of DNA. In the context of genome assembly, contigs are created by piecing together shorter sequences, called reads, that have been obtained from sequencing technology.</p>
</div>
<p><strong>Running QUAST on command line</strong></p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a>quast.py<span class="w"> </span>assembly.fasta
</code></pre></div>
<p>QUAST offers various options and parameters to customize the analysis and output. You can specify different options to generate specific reports or change the output format. For example, you can use the -o flag to specify a different output directory, or use -R to provide a reference genome for alignment if available.</p>
<p>Here's an example of a more customized command:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a>quast.py<span class="w"> </span>-o<span class="w"> </span>custom_output_folder<span class="w"> </span>-R<span class="w"> </span>reference.fasta<span class="w"> </span>-g<span class="w"> </span>gene_annotation.gff<span class="w"> </span>assembly.fasta
</code></pre></div>
<p>This command specifies a custom output folder, uses a reference genome for alignment, and provides a gene annotation file for additional analysis.</p>
<p>Please refer to the QUAST documentation for a full list of available options and their descriptions. The specific options and settings you use may depend on your analysis goals and the characteristics of your data.</p>
<h2 id="completeness-genome-assembly">Completeness (Genome assembly)</h2>
<p>Genome completeness refers to the extent to which a sequenced genome accurately represents the full genetic material of an organism. It is a measure of how thoroughly the genome has been sequenced and assembled, reflecting the presence of all expected genes, sequences, and structural elements.You can assess genome completeness by comparing your assembly to a reference genome, if available. You can also assess genomes via the number of essential genes for that organism. How can we assess completeness?</p>
<ul>
<li>Compare to reference genome (How to find a reference genome? Start with <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">web BLAST</a>)</li>
<li>Use QUAST to compare to reference genome via different metrics, or align the two genomes and inspect via <a href="https://darlinglab.org/mauve/mauve.html">Mauve</a> or <a href="https://www.sanger.ac.uk/tool/artemis/">Artemis</a>.</li>
<li>Assume a genome should have single copy essential genes:<ul>
<li><a href="https://github.com/tseemann/mlst">MLST</a> intact?</li>
<li><a href="https://busco.ezlab.org/">BUSCO</a> panel</li>
<li><a href="https://ecogenomics.github.io/CheckM">CheckM</a> panel</li>
</ul>
</li>
</ul>
<p><strong>Running BUSCO</strong></p>
<p>Here is some example code to run BUSCO on our example data.</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a>conda activate week2 
<a id="__codelineno-2-2" name="__codelineno-2-2" href="#__codelineno-2-2"></a>conda install -c conda-forge -c bioconda busco=5.5.0
<a id="__codelineno-2-3" name="__codelineno-2-3" href="#__codelineno-2-3"></a>busco --list-datasets
<a id="__codelineno-2-4" name="__codelineno-2-4" href="#__codelineno-2-4"></a>wget https://mmbdtp.github.io/seq-analysis/long_assembly.fasta
<a id="__codelineno-2-5" name="__codelineno-2-5" href="#__codelineno-2-5"></a>busco -i long_assembly.fasta  --out assembly-busco  --mode genome -l bacteria_odb10
<a id="__codelineno-2-6" name="__codelineno-2-6" href="#__codelineno-2-6"></a>cat assembly-busco/short_summary.specific.bacteria_odb10.test-busco.txt
</code></pre></div>
<h2 id="contamination-genome-assembly">Contamination (Genome assembly)</h2>
<p>Contamination in genome assembly refers to the presence of extraneous DNA sequences from sources other than the target organism in the final assembled genome. These unwanted sequences can originate from various sources and can significantly compromise the accuracy and reliability of the genome assembly. Some common reasons for contamination in sequencing data include:</p>
<ul>
<li>Sample Cross-Contamination</li>
<li>Contaminated Reagents and Kits</li>
<li>Environmental Contamination</li>
<li>Human Contamination</li>
<li>Cross-Talking in Multiplexed Sequencing</li>
<li>Lab Equipment Contamination</li>
<li>Library Preparation and PCR Artifacts</li>
<li>Sample Mix-Up</li>
</ul>
<p>We can use Kraken2 in the same way that we used it for sequence reads. </p>
<p><strong>Using Kraken2</strong>
Here is some example code to run Kraken2.</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a>conda activate my_env 
<a id="__codelineno-3-2" name="__codelineno-3-2" href="#__codelineno-3-2"></a>conda install -y kraken2 
<a id="__codelineno-3-3" name="__codelineno-3-3" href="#__codelineno-3-3"></a>kraken2 du -h -d1 /shared/public/db/kraken2
<a id="__codelineno-3-4" name="__codelineno-3-4" href="#__codelineno-3-4"></a>kraken2 --threads 8 --db /shared/public/db/kraken2/k2_standard_08gb/ --output long.hits.txt --report long.report.txt  --use-names long_assembly.fasta
</code></pre></div>
<h2 id="correctness-genome-assembly">Correctness (Genome assembly)</h2>
<p>Assess the accuracy of your assembly by checking for misassemblies, such as structural errors, inversions, or translocations. Visualization tools like Artemis or Bandage can help identify such issues. Effectively we are trying to assess, is the genome assembly what we expect? How can we assess correctness?</p>
<ul>
<li>Assembly free from errors</li>
<li>Mis-joins</li>
<li>Collapsed repeats</li>
<li>Duplication artefacts </li>
<li>False SNPs, InDels</li>
<li>Comparison to Other Assemblies: If other assemblies of the same species are available, compare your assembly to them to identify any discrepancies. Ideally to well known reference genome.</li>
<li>Map original reads back to assembled contigs</li>
<li>Evaluation of Plasmids: If the bacterium has plasmids, confirm that they are correctly assembled and identify their sequences.</li>
<li>Structural rearrangement tools - <a href="https://github.com/quadram-institute-bioscience/socru">Socru</a></li>
<li>Try looking at the graph in <a href="https://rrwick.github.io/Bandage/">Bandage</a></li>
</ul>
<h2 id="bonus-circumstantial-genome-assembly">BONUS: Circumstantial (Genome assembly)</h2>
<p>These are not direct evidence of a good genome, but can be reassuring. Here are some circumstanial evidence of a good genome:</p>
<ul>
<li><strong>GC Content:</strong> Verify that the GC content of your assembly matches the expected GC content for the species. Significant deviations could indicate contamination or assembly errors.</li>
<li><strong>Repeat Content:</strong> Assess the presence and handling of specific repetitive elements. High levels of repeats may lead to fragmented assemblies or misassemblies.</li>
<li><strong>Quality of Reads:</strong> Examine the quality of the raw sequencing reads to ensure that they are of high quality, with minimal errors or biases.</li>
<li><strong>Coverage Depth:</strong> Evaluate the coverage depth across the genome. Uniform coverage indicates a more reliable assembly.</li>
<li><strong>Visualization:</strong> Use genome visualization tools like Artemis, IGV, or Tablet to visually inspect the assembly and confirm its quality.</li>
</ul>
<p>Remember that the quality of your bacterial genome assembly may also depend on the sequencing technology used, the software and parameters employed for assembly, and the quality of the source DNA. Careful evaluation and validation of your assembly are essential for accurate results.</p>
<p>Let's apply these criteria to some sample data in <a href="/quality-control/50-assembly-qc-exercise">Practical - Genome assembly QC</a>. </p>







  
  






                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "../..", "features": [], "search": "../../assets/javascripts/workers/search.b8dbb3d2.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.081f42fc.min.js"></script>
      
    
  </body>
</html>